.github/workflows/verify.yml

```diff
@@ -0,0 +1,74 @@
+name: Verify Submodule Pull Request
+
+on:
+  workflow_call:
+    inputs:
+      repo:
+        required: false
+        type: string
+        default: Lombiq/Open-Source-Orchard-Core-Extensions
+
+env:
+  PULL_REQUEST_TITLE: ${{ github.event.pull_request.title }}
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  parent-pull-request-exists:
+    runs-on: ubuntu-latest
+    name: Parent PR Exists
+    steps:
+      - name: Print PR Title
+        run: echo $Env:PULL_REQUEST_TITLE
+
+      - name: Add Scripts
+        run: |
+          $code = @'
+          <#
+          .Synopsis
+          Fails the current step with the provided message. Same as "core.setFailed()" from the official toolkit
+          (https://github.com/actions/toolkit).
+          #>
+          function Set-Failed([Parameter(Mandatory=$True)] [string] $Message)
+          {
+              echo "::error::$Message"
+              exit 1
+          }
+
+          <#
+          .Synopsis
+          Returns $True if any item is coming from the pipe.
+          #>
+          function Any-Object([Parameter(Mandatory=$True, ValueFromPipeline = $True)] $Input)
+          {
+              foreach($item in $Input)
+              {
+                  return $True
+              }
+
+              return $False
+          }
+          '@
+
+          $ProfileDirectory = [System.IO.Path]::GetDirectoryName($Profile)
+          [System.IO.Directory]::CreateDirectory($ProfileDirectory)
+
+          echo $code >> $Profile
+
+      - name: Current PR Title Includes Issue Code
+        run: |
+          if ($Env:PULL_REQUEST_TITLE -match '^\s*\w+-\d+\s*:') { exit 0 }
+          Set-Failed 'The pull request is not in the right format. Please start with the issue code followed by a colon e.g. "PROJ-123: My PR Title".'
+
+      - name: Search Parent Repository for Matching PR
+        run: |
+          $issueCode = $Env:PULL_REQUEST_TITLE -replace '^\s*(\w+-\d+)\s*:.*$', '$1'
+
+          $url = 'https://api.github.com/repos/Lombiq/Open-Source-Orchard-Core-Extensions/pulls?state=open&per_page=100'
+          $titles = curl -H 'Accept: application/vnd.github.v3+json' $url | ConvertFrom-Json | % { $_.title }
+
+          $lookFor = "${issueCode}:"
+          if ($titles | ? { $_ -and $_.Contains($lookFor) } | Any-Object) { exit 0 }
+          Set-Failed "Couldn't find an Open-Source-Orchard-Core-Extensions pull request whose title starts with `"$lookFor`"."
```

Comment:
I was asking for the tooling you're using because I remember that we wanted to apply some code styling convention to Powershell scripts by using [VS]Code with some Powershell plugin, and that would discourage (or prohibit) the usage of `% { }` and `? { }` in favor of `Select-Object` and `Where-Object`. Those are arguably longer but they allow for better understanding in the long run.

Reply:
OK, I think we should extract our PS scripts into files and just call them from the workflow [like here](https://www.ntweekly.com/2021/07/28/run-a-powershell-script-on-github-actions/). That way, we can adhere to coding conventions in those scripts and enjoy proper syntax highlighting! 🥳

Even more advantages:
1. Room for proper comments.
2. Shorter workflow files.

Reply:
> and that would discourage (or prohibit) the usage of `% { }` and `? { }` in favor of `Select-Object` and `Where-Object`

If there is any tooling that doesn't let me using `%` and `?` I want nothing to do with it. I've drawn a line in the sand in front of the wagons that I've encircled around this hill that I will die on. 😁 Luckily the linter actually [has a whitelist](https://docs.microsoft.com/en-us/powershell/utility-modules/psscriptanalyzer/rules/avoidusingcmdletaliases?view=ps-modules#alias-allowlist). Just like we use `hca` in Orchards we should permit some super common abbreviations like those two in pwsh. I'm willing to use `Get-Childitem` instead of `ls`, but everything has its limits.

I'm also moderately against the [PSUseApprovedVerbs](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.2) rule. This isn't an old school text adventure game to only use pre-approved verbs. It takes too long to figure out what verb to shoehorn into a custom function. (Like what should I call `Any-Object`? The best I could come up with is `Measure-NotEmpty` which just rolls off the tongue... directly into the garbage can!) However I think this should be ironed out in a separate issue and with open discussion in Teams.

I've fixed the rest based on the feedback I've got when copying the steps into Code.

> OK, I think we should extract our PS scripts into files and just call them from the workflow [like here](https://www.ntweekly.com/2021/07/28/run-a-powershell-script-on-github-actions/)

Doesn't work, we've already tried this with bash [in a previous issue](https://github.com/Lombiq/GitHub-Actions/pull/2#discussion_r796065264). What you linked works fine for local workflows, but once you try to call into a remote workflow ([like we do here](https://github.com/Lombiq/Orchard-Data-Tables/pull/103/files#diff-b32b798d8ec577b859922a84b90dfe871b19fad404cc6fa7a6fe163f079fbcbfR12)) the situation changes. The content of the remote workflow's repository (this repo) doesn't seem to be accessible. I agree that it would be a good thing if it was possible.

Reply:
You must be kidding me 😢

Reply:
Could [this](https://docs.github.com/en/actions/learn-github-actions/contexts#:~:text=will%20be%20actionscheckout2.-,github.action_path,-string) be of help, maybe? What do you think?

Reply:
I don't see how. Contexts are variables, not files.

Reply:
I thought like this maybe:

```yaml
run: |
    ${{ github.action_path }}/myscript.ps1
```

Reply:
Despite the misleading repo name, these are not actions but "[reusable workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows)". Each action must have a dedicated repo and they have access to the whole thing as a separate directory on the VM. But that would be a huge overkill for something of this complexity.

I have tried and indeed it doesn't work because `github.action_path` is empty: https://github.com/Lombiq/Orchard-Data-Tables/runs/6794141499?check_suite_focus=true

Reply:
Thanks for trying this out. My fingers are itching to dive into this on my own, because it seems unacceptable that extracting scripts into files should not be possible. This should be the way to go, imo.

Anyway, for this issue, let's drop it.

Reply:
I guess reusing workflows is considered a niche and ill-supported use-case much like using submodules in git. (I'm still searching for a free git client that does submodules well on Linux.)

Reply:
Just as a sidenote: While [we could publish our automation code as actions](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2545614855/Writing+GitHub+Actions+workflows#Ideas-for-the-future), this isn't necessary just for our own stuff, and [we can have any number of actions in a single repo](https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2545614855/Writing+GitHub+Actions+workflows#:~:text=runners%20also%20do.-,DRY%20and%20composition,-%3A%20You%20can%20factor).